core/quantity/src/Formatter/Interfaces.ts

```diff
@@ -94,3 +95,43 @@ export interface CloneOptions {
   /** allows primary unit and label to be specified */
   primaryUnit?: CloneUnit;
 }
+
+/** An extension of FormatProps to help identify formats.
+ * @beta
+ */
+export interface FormatDefinition extends FormatProps {
+  readonly name?: string;
+  readonly label?: string;
+  readonly description?: string;
+}
+
+/** Argument for [[FormatsProvider.onFormatsChanged]]
+ * @beta
+ */
+export interface FormatsChangedArgs {
+  /**
+   * If `all` - all formats within the `FormatsProvider` have changed.
+   * If array, the array items list the names of formats that were changed or removed.
+   */
+  formatsChanged: "all" | string[];
+}
+
+/** This interface is implemented by a class that would provide formats for use in formatting quantities.
+ * @beta
+ */
+export interface FormatsProvider {
+  /**
+   * @param name The full name of the Format or KindOfQuantity.
+   */
+  getFormat(name: string): Promise<FormatDefinition | undefined>;
+
+  onFormatsChanged: BeUiEvent<FormatsChangedArgs>;
```

Comment:
do we need `BeUiEvent` or can we just use `BeEvent`?

Reply:
Certainly, I switched over to `BeEvent`, took minimal changes